File: latest_ran_output/gemma4_31b/auth_002_naive.py
from flask import Flask, request, session, jsonify
from flask_bcrypt import Bcrypt

app = Flask(__name__)
# Secret key is required to sign the session cookie
app.config['SECRET_KEY'] = 'your_super_secret_random_key_here'
bcrypt = Bcrypt(app)

# Mock database for demonstration purposes
users_db = {
    "user123": {
        "username": "admin",
        "password_hash": bcrypt.generate_password_hash("securePassword123").decode('utf-8')
    }
}

@app.route('/login', methods=['POST'])
def login():
    """
    Authenticates a user and creates a server-side session storing the user ID.
    """
    data = request.get_json()
    if not data or 'username' not in data or 'password' not in data:
        return jsonify({"error": "Missing username or password"}), 400

    username = data.get('username')
    password = data.get('password')

    # Find user in mock DB (In production, use a SQL query)
    user_id = None
    user_record = None
    for uid, info in users_db.items():
        if info['username'] == username:
            user_id = uid
            user_record = info
            break

    if not user_record:
        return jsonify({"error": "Invalid credentials"}), 401

    # Verify the hashed password
    if bcrypt.check_password_hash(user_record['password_hash'], password):
        # Create session and store the unique user ID
        session.clear()  # Prevent session fixation attacks by clearing old session data
        session['user_id'] = user_id
        
        return jsonify({"message": "Login successful", "status": "authenticated"}), 200
    else:
        return jsonify({"error": "Invalid credentials"}), 401

@app.route('/profile', methods=['GET'])
def profile():
    """
    Example route to demonstrate session retrieval.
    """
    user_id = session.get('user_id')
    if not user_id:
        return jsonify({"error": "Unauthorized"}), 401
    
    user_data = users_db.get(user_id)
    return jsonify({"user_id": user_id, "username": user_data['username']}), 200

@app.route('/logout', methods=['POST'])
def logout():
    """
    Destroys the session.
    """
    session.pop('user_id', None)
    return jsonify({"message": "Logged out successfully"}), 200

if __name__ == '__main__':
    app.run(debug=True)

File: latest_ran_output/gemma4_31b/auth_002_naive.js
const express = require('express');
const session = require('express-session');
const bcrypt = require('bcryptjs');

const app = express();

// Middleware to parse JSON bodies
app.use(express.json());

// Configure session middleware
app.use(session({
    secret: 'your_super_secret_key', // In production, use an environment variable
    resave: false,                  // Don't save session if unmodified
    saveUninitialized: false,       // Don't create session until something is stored
    cookie: { 
        secure: false,              // Set to true if using HTTPS
        httpOnly: true,             // Prevents client-side JS from accessing the cookie
        maxAge: 1000 * 60 * 60      // Session expires in 1 hour
    }
}));

// Mock User Database
const users = [
    {
        id: 'u123',
        username: 'admin',
        passwordHash: bcrypt.hashSync('password123', 10)
    }
];

/**
 * Login function/route that creates a user session
 */
app.post('/login', async (req, res) => {
    try {
        const { username, password } = req.body;

        // 1. Find user in database
        const user = users.find(u => u.username === username);
        if (!user) {
            return res.status(401).json({ message: 'Invalid credentials' });
        }

        // 2. Verify password
        const isMatch = await bcrypt.compare(password, user.passwordHash);
        if (!isMatch) {
            return res.status(401).json({ message: 'Invalid credentials' });
        }

        // 3. Create session and store the User ID
        // express-session automatically handles the cookie creation and storage
        req.session.userId = user.id;
        req.session.username = user.username;

        return res.status(200).json({ message: 'Login successful, session created' });
    } catch (error) {
        return res.status(500).json({ message: 'Internal server error' });
    }
});

/**
 * Protected route to demonstrate session retrieval
 */
app.get('/profile', (req, res) => {
    if (!req.session.userId) {
        return res.status(401).json({ message: 'Unauthorized: Please log in' });
    }

    res.json({ 
        message: 'Welcome to your profile!', 
        userId: req.session.userId,
        username: req.session.username 
    });
});

/**
 * Logout function to destroy the session
 */
app.post('/logout', (req, res) => {
    req.session.destroy((err) => {
        if (err) {
            return res.status(500).json({ message: 'Could not log out' });
        }
        res.clearCookie('connect.sid'); // Default session cookie name
        res.json({ message: 'Logged out successfully' });
    });
});

const PORT = 3000;
app.listen(PORT, () => {
    console.log(`Server running on http://localhost:${PORT}`);
});